Nos/Models/JSONEvent.swift

```diff
@@ -94,4 +94,20 @@ struct MetadataEventJSON: Codable {
     private enum CodingKeys: String, CodingKey {
         case displayName = "display_name", name, about, picture
     }
+    
+    var dictionary: [String: String] {
+        [
+            "display_name": displayName ?? "",
+            "name": name ?? "",
+            "about": about ?? "",
+            "picture": picture ?? "",
```

Comment:
I think there is a problem with publishing this dictionary, which is that it will overwrite any fields that the user has added to their profile from other clients. Some clients let you put in fields like NIP-05, personal website, etc.

I think we could add a field to Author that's like "unsupportedMetadata" or something that is a dictionary with these keys and values that we don't recognize. Then when we are creating this json we can add in those keys and values so they don't get clobbered. Similar to what we are doing with `Event.allTags`.

Reply:
Ok. I've updated it to store the raw metadata on parse. Then on publish, it adds any keys from that json into the event that gets published. I tested this using the website field with Damus and it seems to work.